### ISSUE TITLE
Model Overfitting and Poor Performance Due to Lack of Data Shuffling During Training

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#1444]
Model overfitting and lack of data shuffling during training results in poor performance on unseen data, dataset not utilizing EpochShuffleDataset causing data to be presented to the model in the same order at each epoch.
